bokeh/tile_providers.py

```diff
@@ -232,6 +241,15 @@ def get_provider(self, provider_name):
             # This allows `get_provider(CARTODBPOSITRON)` to work
             return WMTSTileSource(url=provider_name.url, attribution=provider_name.attribution)
 
+        if xyzservices:
+            if isinstance(provider_name, xyzservices.TileProvider):
+                return WMTSTileSource(
+                    url=provider_name.build_url(scale_factor="@2x"),
+                    attribution=provider_name.html_attribution,
+                    min_zoom=provider_name.get("min_zoom", 0),
+                    max_zoom=provider_name.get("max_zoom", 30),
+                )
```

Comment:
generally speaking we follow this pattern for less nesting:
```suggestion
        if xyzservices and isinstance(provider_name, xyzservices.TileProvider):
            return WMTSTileSource(
                url=provider_name.build_url(scale_factor="@2x"),
                attribution=provider_name.html_attribution,
                min_zoom=provider_name.get("min_zoom", 0),
                max_zoom=provider_name.get("max_zoom", 30),
            )
```